### ISSUE TITLE
Generalize handling of atmospheric field names in derived quantities

### ISSUE THREAD

**Issue by jpdunc23** [ai2cm/ace#181]
I see three options:

1. Add some sort of config parameter for specifying which `in_names` match which standard names used for derived quantities. Here's an experimental PR which takes that approach: https://github.com/ai2cm/full-model/pull/261
2. Map `in_names` to some set of standardized names, e.g. change `in_names` to a dictionary along the lines of:
```
# train-config.yaml
...
stepper:
  ...
  in_names:
    PS: surface_pressure
    FLUT: toa_upward_longwave_flux
```
3. Handle the mapping `CLIMATE_FIELD_NAME_PREFIXES` in dataclasses like `FV3GFSData` and `E3SMV2Data`. The former is currently only used for testing and the latter doesn't exist yet. But I think it makes sense to have a climate dataset typing abstraction for each new dataset that we add (e.g., could also have `ERA5Data`), and maybe each of these could inherit from `ClimateData`.

I personally think option 3 is nicest.

**Comment by oliverwm1** [comment]
I agree something like option 3 would be best. This could also encode physical constants which vary slightly between models.

**Comment by mcgibbon** [comment]
Yes, something like 3. For the API exposed outside of the data loaders, I'd suggest including this as a `to_climate_data(data: Mapping[str, torch.Tensor]) -> ClimateData` method on the `Dataset` class. This would likely involve storing a climate_field_name_prefixes privately inside the `Dataset`.

Also note that (3) isn't exclusive of (1) and (2), either of these could in theory be included as DataLoaderParams configuration. I think that's where this config should live if it exists, rather than in the stepper. I doubt we need it, as our datasets are slow moving enough that we've always been doing PRs to update to the new one, but it's an option.

**Comment by mcgibbon** [comment]
Is there a use case that requires this change? It's hard for me to compare its priority to other issues without knowing the underlying problem driving this task.

**Comment by gideonite** [comment]
I'm not an expert on this, but I believe that the field names differ between FV3 and E3SM. This is an indicator that we do want this code to be applicable to datasets other than FV3 which may have different field names. @jpdunc23 is more expert on this. Specifically, it would be great to confirm that indeed the field names are different in the E3SM dataset.

**Comment by jpdunc23** [comment]
@mcgibbon @gideonite yes, some field names differ in E3SM, but not all of them. PREsfc is called PS, which is the reason I encountered bug https://github.com/ai2cm/ace/issues/178.  See the [fourcastnet-perlmutter `train-config.yaml`](https://github.com/ai2cm/vcm-workflow-control/blob/master/examples/fourcastnet-perlmutter/config-train-and-inference/train-config.yaml) example for (most of/all of by EOD tomorrow) the E3SMv2 field names we use.

**Comment by mcgibbon** [comment]
@gideonite Can we resolve this much more simply by defining a (length 2) list of potential names for each physical variable, and returning the one (if any) that is present in the data? I'm not sure the complexity of a fully configurable naming system is worth it when we have full control over our 2-3 datasets.

**Comment by gideonite** [comment]
@mcgibbon That does seem like a simpler solution. We have full control over the datasets and the code at this point, so why not continue hard-coding but make it slightly more general.

**Comment by mcgibbon** [comment]
OK sounds good. Feel free to pick up this task, from the config it looks like for now the only variable needing this is `"PS"`.

**Comment by jpdunc23** [comment]
Sounds fine to me. Approach 3 would be nice as a centralized abstraction for this sort of thing along with other dataset-specific traits (dimension names, calendar type, metadata, step size, vertical coords, etc.) But probably not worth the effort unless you anticipate a need.

**Comment by oliverwm1** [comment]
Was this resolved by ai2cm/full-model#445?

**Comment by gideonite** [comment]
It looks like this is extensible and can handle FV3 and E3SM. So I think we can close

**Comment by mcgibbon** [comment]
Resolved by ai2cm/full-model#445.
